### PR TITLE
Added test for the personRepository using mockito and adding necessery  dependency to pom.xml part of #10

### PR DESCRIPTION
**Pull request by Charlottehenriksson**
Added test for the personRepository using mockito and adding necessary dependency to pom.xml
Including only tests for PersonRepository and added dependency to pom.xml

### CODE REVIEW

File: src/test/java/se/iths/jakartaeelab1/repository/PersonRepositoryTest.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    @Mock
+    private EntityManager entityManager;
+
+    @InjectMocks
+    private PersonRepository personRepository;
+
+    @Test
+    void createPersonTest() {
+        Person person = new Person();
+        doNothing().when(entityManager).persist(any());
```
Comment (Danback) [inline src/test/java/se/iths/jakartaeelab1/repository/PersonRepositoryTest.java:30]: Is this line needed? persist here should do nothing by default when it's mocked.
I could be wrong though.

@Test
void createPersonTest() {
    Person person = new Person();

    personRepository.createPerson(person);

    verify(entityManager).persist(person); 
}

**Review by Win-ther — CHANGES_REQUESTED** [review]
The tests give full coverage and seems to test that the methods have been called correctly.
There is some changes I would like to see though.

File: pom.xml [hunk 2]
```diff
@@ -35,8 +35,27 @@
...
+            <artifactId>mockito-core</artifactId>
+            <version>5.10.0</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.mockito</groupId>
+            <artifactId>mockito-junit-jupiter</artifactId>
+            <version>5.10.0</version>
+            <scope>test</scope>
+        </dependency>
```
Comment (Win-ther) [inline pom.xml:49]: Mockito Version 5.11.0 seems to be out as of 1/3-2024, maybe we should update it to the latest?

File: src/test/java/se/iths/jakartaeelab1/repository/PersonRepositoryTest.java (hunk 1 above)
Comment (Win-ther) [inline src/test/java/se/iths/jakartaeelab1/repository/PersonRepositoryTest.java:30]: The test seems to be working even without thelling persist to do nothing, is this part neccesary?

File: src/test/java/se/iths/jakartaeelab1/repository/PersonRepositoryTest.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        personRepository.createPerson(person);
+
+        verify(entityManager, times(1)).persist(person);
+    }
+
+    @Test
+    void readAllPersonTest() {
+        List<Person> personList = Arrays.asList(new Person(), new Person());
+
+        jakarta.persistence.TypedQuery<Person> typedQueryMock = mock(jakarta.persistence.TypedQuery.class);
```
Comment (Win-ther) [inline src/test/java/se/iths/jakartaeelab1/repository/PersonRepositoryTest.java:41]: This is weird, why have you chosen to not just import jakarta.persistence.TypedQuery?

File: src/test/java/se/iths/jakartaeelab1/repository/PersonRepositoryTest.java [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+        assertNotNull(result);
+        assertEquals(personList, result);
+    }
+
+    @Test
+    void findPersonByIdTest() {
+        UUID id = UUID.randomUUID();
+        Person person = new Person();
+        when(entityManager.find(eq(Person.class), eq(id))).thenReturn(person);
```
Comment (Win-ther) [inline src/test/java/se/iths/jakartaeelab1/repository/PersonRepositoryTest.java:58]: The eq()-method calls seems unnecessary, it works fine when passing the values directly. SonarLint even comments on it, why have you chosen to use eq() here?

File: src/test/java/se/iths/jakartaeelab1/repository/PersonRepositoryTest.java [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+        assertNotNull(result);
+        assertEquals(updatedPerson, result);
+        verify(entityManager, times(1)).merge(updatedPerson);
+    }
+
+    @Test
+    void deletePersonTest() {
+        UUID id = UUID.randomUUID();
+        Person person = new Person();
+        when(entityManager.find(eq(Person.class), eq(id))).thenReturn(person);
```
Comment (Win-ther) [inline src/test/java/se/iths/jakartaeelab1/repository/PersonRepositoryTest.java:82]: The eq()-method calls seems unnecessary, it works fine when passing the values directly. SonarLint even comments on it, why have you chosen to use eq() here?

File: src/test/java/se/iths/jakartaeelab1/repository/PersonRepositoryTest.java [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+
+    @Test
+    void updatePersonTest() {
+        Person updatedPerson = new Person();
+        when(entityManager.merge(any())).thenReturn(updatedPerson);
+
+        Person result = personRepository.updatePerson(updatedPerson);
+
+        assertNotNull(result);
+        assertEquals(updatedPerson, result);
```
Comment (Win-ther) [inline src/test/java/se/iths/jakartaeelab1/repository/PersonRepositoryTest.java:74]: This assert seems unnecessary aswell, if result is equals to updatedPerson. Is it not by default not null?

File: src/test/java/se/iths/jakartaeelab1/repository/PersonRepositoryTest.java [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+    @Test
+    void findPersonByIdTest() {
+        UUID id = UUID.randomUUID();
+        Person person = new Person();
+        when(entityManager.find(eq(Person.class), eq(id))).thenReturn(person);
+
+        Person result = personRepository.findPersonById(id);
+
+        assertNotNull(result);
+        assertEquals(person, result);
```
Comment (Win-ther) [inline src/test/java/se/iths/jakartaeelab1/repository/PersonRepositoryTest.java:63]: This assert seems unnecessary aswell, if result is equals to person. Is it not by default not null?

File: src/test/java/se/iths/jakartaeelab1/repository/PersonRepositoryTest.java [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+        when(entityManager.createQuery(anyString(), eq(Person.class)))
+                .thenReturn(typedQueryMock);
+
+        when(typedQueryMock.getResultList())
+                .thenReturn(personList);
+
+        List<Person> result = personRepository.readAllPerson();
+
+        assertNotNull(result);
+        assertEquals(personList, result);
```
Comment (Win-ther) [inline src/test/java/se/iths/jakartaeelab1/repository/PersonRepositoryTest.java:51]: This assert seems unnecessary aswell, if result is equals to personList. Is it not by default not null?

**Review by Win-ther — CHANGES_REQUESTED** [review]

**Comment by Win-ther** [comment]
You have some conflicts in your pom.xml still

**Review by altmannk — APPROVED** [review]
LGTM 👍🏽

**Review by Win-ther — APPROVED** [review]
Tests looks good now and gives full coverage for the PersonRepository-class. Good job! :)

**Pull request merged**